design/external-authorization-design.md

```diff
@@ -0,0 +1,670 @@
+# External Authorization Support
+
+Status: Reviewing
+
+## Abstract
+This document describes a design for performing request authorization for virtual hosts hosted by Contour.
+
+## Background
+
+- [#432][5] Provide support for integrating with an external auth service.
+- [#2459][3] External auth background and goals.
+- [#2325][9] Failures in External services configured in envoy should be visible to the application developer.
+
+## Goals
+- Allow operators to integrate existing authorization services with Contour.
+- Allow arbitrary types of authentication and authorization to be supported.
+- Decouple Contour from external authorization, so it can evolve at an independent rate.
+- Integrate cleanly with the HTTPProxy API.
+
+## Non Goals
```

Comment:
I think it'd be useful to explicitly call out somewhere in Goals or Non-Goals that this design proposes the framework, but does not propose auth server implementations for any specific auth protocols, and that those will also need to be implemented before a user can actually make use of external auth.

Reply:
This topic is discussed later in the doc, but I suppose we could add it here too.